### PR TITLE
add devenv files

### PR DESCRIPTION
**Pull request by ss2165**

### CODE REVIEW

File: devenv.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ pkgs, ... }:
+
+{
+  # https://devenv.sh/basics/
+
+  # https://devenv.sh/packages/
+  # manually set rust packages rather than use devenv language support because
+  # it doesn't seem to be up to date for macos yet (link error)
+  packages = with pkgs; [ cargo rustc rust-analyzer rustfmt clippy ];
```
Comment (aborgna-q) [inline devenv.nix:9]:
```suggestion
  packages = with pkgs; [ cargo rustc rust-analyzer rustfmt clippy miri ];
```

**Review by aborgna-q — APPROVED** [review]
nice!

Perhaps we should mention this in the README and/or a CONTRIBUTING file?
(Can be a sepparate issue)

File: devenv.nix (hunk 1 above)
Comment (ss2165) [reply devenv.nix:9]: do we need miri for local?

Comment (aborgna-q) [reply devenv.nix:9]: If it's for the dev environment, I'd say it's one of the normal tools to have.

File: devenv.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    cargo --version
+  '';
+
+  # https://devenv.sh/languages/
+  # https://devenv.sh/reference/options/#languagesrustversion
+  # languages.rust.enable = true;
+
+  # https://devenv.sh/pre-commit-hooks/
+  pre-commit.hooks.clippy.enable = true;
+  pre-commit.hooks.rustfmt.enable = true;
```
Comment (aborgna-q) [inline devenv.nix:26]: 👍

**Review by aborgna-q — APPROVED** [review]

**Pull request merged**